### PR TITLE
Output true srcset, not data-srcset

### PR DESCRIPTION
**Pull request by ZachGawlik**
Completes [CMS-38077](https://jira.squarespace.net/browse/CMS-38077) . The `|image-srcset` formatter will now return the standard `srcset` attribute to be picked up by [modern browsers](https://caniuse.com/srcset). ImageLoader will be updated to operate on this attribute for unsupported browsers rather than `data-srcset`.

Ahead of this work, it was determined that template-engine is not yet processing any JSON-T that uses the image-srcset formatter.

### CODE REVIEW

**Review by nhassan — APPROVED** [review]

**Review by phensley — APPROVED** [review]

**Review by kauffecup — APPROVED** [review]

**Comment by phensley** [comment]
@ZachGawlik I'll hold off on merging until the backend compiler changes are also ready. I assume we need to hold on releasing until the imageloader changes are made, correct?

**Review by Danjohnsonnj — COMMENTED** [review]
This is exactly as we discussed, if I recall correctly. The template developer will use both the typical `image-meta` formatter with this one to achieve the correct markup. I only wonder if that has been tested? Since this formatter is not typically to be used on its own but rather in the manner described, should we create a specific test for this use-case?

**Comment by Danjohnsonnj** [comment]
@phensley, re: 
> @ZachGawlik I'll hold off on merging until the backend compiler changes are also ready. I assume we need to hold on releasing until the imageloader changes are made, correct?

Since there are no current usages of the srcset compiler, there is no need to update ImageLoader until these changes are merged. At that point, IL will be updated to account for the inclusion of `srcset` instead of `data-srcset`.

**Comment by phensley** [comment]
Anyone know who is handling the backend compiler implementation?

**Review by georgezzhang — APPROVED** [review]

**Review by mattlyons0 — APPROVED** [review]

**Comment by phensley** [comment]
This is released in version `2.6.0`

**Pull request merged**
